web-client/iron-svelte-client/src/lib/login/login.svelte

```diff
@@ -140,38 +143,44 @@
 
         const config = configBuilder.build();
 
-        from(userInteraction.connect(config))
-            .pipe(
-                catchError((err) => {
-                    toast.set({
-                        type: 'info',
-                        message: err.backtrace(),
-                    });
-                    return of(null);
-                }),
-                filter((result) => result !== null && result !== undefined), // Explicitly checking for null/undefined
-            )
-            .subscribe((info: NewSessionInfo | null) => {
-                if (info != null && info.initialDesktopSize !== null) {
-                    toast.set({
-                        type: 'info',
-                        message: 'Success',
-                    });
-                    currentSession.update((session) =>
-                        Object.assign(session, {
-                            sessionId: info.sessionId,
-                            desktopSize: info.initialDesktopSize,
-                            active: true,
-                        }),
-                    );
-                    showLogin.set(false);
-                } else {
-                    toast.set({
-                        type: 'error',
-                        message: 'Failure',
-                    });
-                }
-            });
+        try {
+            const session_info = await userInteraction.connect(config);
+
+            if (session_info.initialDesktopSize !== null) {
```

Comment:
question: Is it possible for this to be null at all?

Reply:
No. I removed the check